### ISSUE TITLE
CMSSW doesn't work with edge singularity image

### ISSUE THREAD

**Issue by irenedutta23** [cmscaltech/gpuservers#20]
I was trying to run CMSSW with the `edge` image, but it gives the following error:
```
-bash-4.2$ singularity exec --home /storage/user/jbalcas/:/srv --bind /mnt/hadoop --bind /storage --bind /cvmfs --pwd /srv --contain --ipc --pid /storage/group/gpu/software/singularity/ibanks/edge.simg bash
Singularity>
Singularity> source /cvmfs/cms.cern.ch/cmsset_default.sh
Singularity> cd /storage/cc^C
Singularity> cd tmp/
Singularity> ls
Singularity> cmsrel CMSSW_10_2_8
Can't locate Data/Dumper.pm in @INC (@INC contains: /cvmfs/cms.cern.ch/share/lcg/SCRAMV1/V2_2_9_pre11 /cvmfs/cms.cern.ch/share/lcg/SCRAMV1/V2_2_9_pre11/src /usr/local/lib64/perl5 /usr/local/share/perl5 /usr/lib64/perl5/vendor_perl /usr/share/perl5/vendor_perl /usr/lib64/perl5 /usr/share/perl5 .) at /cvmfs/cms.cern.ch/share/lcg/SCRAMV1/V2_2_9_pre11/src/Cache/CacheUtilities.pm line 3.
BEGIN failed--compilation aborted at /cvmfs/cms.cern.ch/share/lcg/SCRAMV1/V2_2_9_pre11/src/Cache/CacheUtilities.pm line 3.
Compilation failed in require at /cvmfs/cms.cern.ch/share/lcg/SCRAMV1/V2_2_9_pre11/src/SCRAM/CMD.pm line 500.
BEGIN failed--compilation aborted at /cvmfs/cms.cern.ch/share/lcg/SCRAMV1/V2_2_9_pre11/src/SCRAM/CMD.pm line 500.
Compilation failed in require at /cvmfs/cms.cern.ch/share/lcg/SCRAMV1/V2_2_9_pre11/src/SCRAM/SCRAM.pm line 35.
BEGIN failed--compilation aborted at /cvmfs/cms.cern.ch/share/lcg/SCRAMV1/V2_2_9_pre11/src/SCRAM/SCRAM.pm line 35.
Compilation failed in require at /cvmfs/cms.cern.ch/share/lcg/SCRAMV1/V2_2_9_pre11/bin/scram line 19.
BEGIN failed--compilation aborted at /cvmfs/cms.cern.ch/share/lcg/SCRAMV1/V2_2_9_pre11/bin/scram line 19.
```

**Comment by vlimant** [comment]
if one adds `-B /usr` to the singularity exec/run command it functions

```
-bash-4.2$ singularity shell -B /usr -B /etc/grid-security/letsencrypt -B /data -B /imdata -B /mnt/hadoop -B /storage -B /cvmfs --nv /storage/group/gpu/software/singularity/ibanks/edge.simg
WARNING: underlay of /etc/grid-security/letsencrypt required more than 50 (124) bind mounts
Singularity> source /cvmfs/cms.cern.ch/cmsset_default.sh
Singularity> cmsrel CMSSW_10_2_8
WARNING: Release CMSSW_10_2_8 is not available for architecture slc7_amd64_gcc820.
         Developer's area is created for available architecture slc7_amd64_gcc700.
WARNING: Developer's area is created for non-production architecture slc7_amd64_gcc700. Production architecture for this release is slc6_amd64_gcc700.
Singularity>
```

I have to figure out what should the perl5 installation be in the singularity image to have the same version as the node. @juztas any hint ?

**Comment by vlimant** [comment]
installing perl-Data-Dumper seems to be doing the trick, will update the edge image accordingly.

**Comment by vlimant** [comment]
fixed with 20f1b5f9b01a145c96b1b5e400e7890c619b1258
